### PR TITLE
added graphical selector for some entities and fixed some issues

### PR DESCRIPTION
**Pull request by Vegita2**
Added New graphical selection widget for some entities:
### Prop
image

### ScalableProp
image

### Enemy
image

### ItemDestruct
image

## other changes
- Updated Phaser to 3.60.0
- Entity copy/paste now uses the clipboard to make it easier working with multiple editor instances #267
- All `Prop` Entities should now be rendered correctly
- Improved `ScalableProp` rendering. `ends` are now rendered correctly.
- Releasing mouse button outside the canvas now properly triggers. Avoids having issues like continue dragging entities even if the mouse button was released

### CODE REVIEW

File: webapp/src/app/services/phaser/entities/entity-manager.ts [hunk 1]
```diff
@@ -355,27 +367,58 @@ export class EntityManager extends BaseObject {
...
+		const json = await navigator.clipboard.readText();
+		let entities: MapEntity[] = [];
+		try {
+			let parsed = JSON.parse(json);
+			if (!Array.isArray(parsed)) {
+				parsed = [parsed];
+			}
+			entities = (parsed as any[]).filter(v => this.isMapEntity(v));
+		} catch (e) {
+			console.warn('could not parse entities from clipboard');
```
Comment (2767mr) [inline webapp/src/app/services/phaser/entities/entity-manager.ts:397]: This should probably also show a snackbar warning

**Review by 2767mr — CHANGES_REQUESTED** [review]
Things I noticed:
* A loading indicator would be nice
* Search should include sub types (with toggle?)
* Perhaps it would look better if this is centered?
  image
* I think this could use some extra margin-right
  image
* There is no good way to see the full name of the character now. In my other projects I usually do `text-overflow: ellipsis` with a tooltip containing the full text. This would clash with the existing tooltips here but maybe we can figure something out.
  image
* Creating a new ScalableProp creates a 1x1 px entity (was already the case before this PR)
* I thought about opening the prop picker after double clicking props but that would probably be a bit much. Maybe opening it after creating a new prop would be nice?


Things I really liked:
* The "transparent texture"
* The size change on hover
* The red outline
* The counter in the top right of the props
* The prop rendering. That one has bothered me for a long time.

**Comment by Vegita2** [comment]
> Perhaps it would look better if this is centered?

Is this possible? the container uses `flex-wrap: wrap;` so i'm not sure how you could center them after the wrapping

> I thought about opening the prop picker after double clicking props but that would probably be a bit much. Maybe opening it after creating a new prop would be nice?

We could always trigger the double click when adding a new entity, but not sure if it's always the first thing you want to do.  I would keep it like it currently is for now and decide later with some user feedback. It could also be toggleable in the settings menu

* [ ] A loading indicator would be nice
* [ ] Search should include sub types (with toggle?)
* [ ] I think this could use some extra margin-right
   image
 * [ ] There is no good way to see the full name of the character now. In my other projects I usually do `text-overflow: ellipsis` with a tooltip containing the full text. This would clash with the existing tooltips here but maybe we can figure something out.
   image
* [x] Creating a new ScalableProp creates a 1x1 px entity (was already the case before this PR)

**Comment by lexisother** [comment]
Can this be made to scroll down to the selected prop/enemy/destructible when you open the modal again?

**Review by 2767mr — APPROVED** [review]

**Comment by Vegita2** [comment]
@lexisother made a new issue for it #277

**Pull request merged**